File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        if(n <= 0)
            return 0;
        if(n <= 2)
            return 1;
        long long int first = 0;
        long long int second = 1;
        long long int third = 1;
        long long int num = 0;
        for(int i = 2; i < n; i++)
        {
            num = first + second + third;
            first = second; 
            second = third;
            third = num;
        }
        return num;
    }

    
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.go

func tribonacci(n int) int {
    mp := make(map[int]int)
    return helper(n, mp)
}

func helper(n int, mp map[int]int) int {
    if n == 0 {
        return 0
    }
    if n == 1 || n == 2 {
        return 1
    }
    if val, ok := mp[n]; ok {
        return val
    }

    mp[n] = helper(n-1, mp) + helper(n-2, mp) + helper(n-3, mp)
    return mp[n]
}